File: src/io/context/LookUpEntity.cpp
#include "io/context/LookUpEntity.h"

LookUpEntity::LookUpEntity():
    key(nullptr),
    value(0),
    data(nullptr)
{
}

LookUpEntity::LookUpEntity(char *key, long value, char *data):
    key(key),
    value(value),
    data(data)
{
}

File: java/src/main/java/vsdk/toolkit/common/dataStructures/LookUpEntity.java
package vsdk.toolkit.common.dataStructures;

public class LookUpEntity<T> {
    public String key; // Key name
    public long value; // Key hash value (for efficiency)
    public T data; // Client data

    public LookUpEntity() {
        key = null;
        value = 0;
        data = null;
    }
}
